### ISSUE TITLE
DNS Filtering prevents connecting to WiFi (not talking about login portals)

### ISSUE THREAD

**Issue by Utini2000** [AdguardTeam/AdguardForAndroid#4797]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.0

### Environment

```markdown
- OS:Android 13 June build
- Device: Pixel 6 Pro
- Firmware: June build

I am using my phone in several different companies and with some of them I can't fully connect to the companies WiFi until I disable DNS protection, connect, re-enable DNS protection. Otherwise it will give me the Wifi symbol with a "!" and disconnect from wifi because of "no internet access".
```

### Root access

- [X] Yes, I have it.

### What filters do you have enabled?

AdGuard Base filter, AdGuard Mobile Ads filter, AdGuard German filter, AdGuard Tracking Protection filter, AdGuard URL Tracking filter, AdGuard Social Media filter, AdGuard Annoyances filter, AdGuard Cookie Notices filter, AdGuard Popups filter, AdGuard Mobile App Banners filter, AdGuard Other Annoyances filter, AdGuard DNS filter, EasyList, EasyList Germany, Others

### What Stealth Mode options do you have enabled?

Send Do-Not-Track signals, Strip URLs of tracking parameters, Self-destruction of third-party cookies

### Issue Details

Steps to reproduce:
Connect to any passwort protected wifi
Leave building and come back a few days later
Phone want to auto-connect to the wifi but disabled the saved connection due to "no internet access".
Disable DNS protection in adguard
WiFi will connect
Enable DNS protection
Stay connected to wifi.

### Expected Behavior

Automatically connect to saved wifi

### Actual Behavior

Doesn't connect to wifi and instead disables it

### Screenshots

<details><summary>Screenshot 1:</summary>

</details>

**Comment by Utini2000** [comment]
This also seems to affect SIM internet (3G, 4G, 5G). I get the "!" symbol in the statusbar and no internet until I disable DNS protection.

**Comment by maxikuzmin** [comment]
@Utini2000 thanks for the long wait!
The problem with switching from SIM card to WIFI and back didn't repeat.
The problem with reconnecting to WIFI after a few days has not yet been repeated. How often does this happen? For how long have you left the buildings? Does it recur with any DNS server or any particular one?

it would be nice to get your logs.

- Enable debug logging:

Settings -> General -> Advanced -> Logging level -> Debug

- Reproduce the issue related to the button behaviour, then remember the exact time when it happened

- Collect logs:

Settings -> General -> Advanced -> Export logs and system info

- Sent this file to [devteam@adguard.com](mailto:devteam@adguard.com):

mention the 4797 number in the subject
specify the exact time when the issue occurred

**Comment by Utini2000** [comment]
@maxikuzmin hi there, I just enabled logging and will try to catch it with debugging enabled.
For testing purposes I already tried cloudflare & quad9 DNS. It happens on both of them.
I am not set to AdguardDNS to try it with that.

Another thing I noticed is that when I switch to "automatic DNS" it will choose/select "DNS server for other cases". I am rooted and on proxy mode btw. 

Do clarify:
I randomly get the "!" symbol next to the "4G or 5G" symbol in my status bar. No internet connection available then.
This sometimes also happens with WiFi.
To fix this, I have to disable and enable DNS protection.
This problem only exists since the current 4.0 version.

Another issue (which has haunted me since older versions) is that I sometimes can't connect to a WiFi until I disabled DNS protection, connect to it, enable DNS protection.
This happens basically with all the WiFi hot spots at my working area (we have like 300 different WiFi switches and it seems like I have to do this on every switch after a certain period I can't really accurately name. Maybe every 1-2 days or so).
This also happens with any DNS server although I am usually always using cloudflare DNS over HTTPS.

**Comment by maxikuzmin** [comment]
@Utini2000 we are not currently experiencing a problem with wifi connection, network change, or loss of internet with DNS connection. I also often use different wifi networks myself, but haven't seen this problem. 
I'll try to check it with Cloudflare DNS. For now, try catching this with debug logs and sending it

**Comment by Utini2000** [comment]
@maxikuzmin I just sent a first log where I wasn't able to get a connection with WiFi. It was fixed after several reconnects. So not exactly the same issue as mentioned before but maybe related.

**Comment by Utini2000** [comment]
@maxikuzmin so I just had the issue, enabled logging, simulated the issue by turning on/off airplane mode and dns protection. Only once I disabled dns protection I was finally able to get network connection. Log file has been sent!

**Comment by Utini2000** [comment]
I have no completely disabled DNS Protection because this feature sadly makes my phone useless.
No problems since disabling it.

**Comment by maxikuzmin** [comment]
@Utini2000 unfortunately I have not been able to reproduce this.
Great! I've got your logs, we'll look into them, it'll take some time.

**Comment by Utini2000** [comment]
@maxikuzmin not sure if it helps but using Android 13 own private DNS feature (e.g. with cloudflare, nextdns or AdguardDNS) works fine.

**Comment by Utini2000** [comment]
@maxikuzmin another follow up:
Using androis private dns feature (Settings -> Network -> DNS) with nextdns (it has the same blocklist as I have in adguard) I noticed that on my office WiFi I can connect to WiFi but won't get an internet connection until I disable & re-enable the private DNS.
I tried to replicate this in my home network and believe it is part of some kind of "dns rebinding protection" in the router/network switch. I know that there are ways to bypass this (e.g. DNSCrypt on Windows / Linux does so).

I don't have any GSM (4G/5G) connection issues though. That is a step up compared to AdGuard still.

**Comment by maxikuzmin** [comment]
@Utini2000 did I understand correctly that this problem also recurs when connecting to DNS server in system settings?

**Comment by Utini2000** [comment]
@maxikuzmin 
On my office Wifi -> Yes
On my home WiFi -> No
On 4G/5G/GSM -> No

I believe my office WiFi has some sort of DNS Rebind Protection but tools like AdGuard should know how to bypass this. 
E.g. DNSCrypt can do this.

**Comment by Utini2000** [comment]
It also seems like Adguard now from time to time disables ad protection. No idea why. Never happend with previous versions and it is probably not related to this issue.

**Comment by maxikuzmin** [comment]
@Utini2000 we will need some more time, thanks for waiting

**Comment by maxikuzmin** [comment]
@Utini2000 could you please specify: 
- Does the problem persist if you switch to local VPN mode? 
- Does turning some other setting on/off solve the problem? This will trigger a restart of the protection

**Comment by Utini2000** [comment]
@maxikuzmin I am currently on vacation in Japan and cannot test. I never tried VPN Mode but I will give it a try next week. Changing any settings / restarting protection fixes the issue most times (although I mostly deactivate adguard, wait a few seconds, activate again to achieve best results).

**Comment by maxikuzmin** [comment]
@Utini2000 okay, I'll wait for news. Have a great vacation!

**Comment by Utini2000** [comment]
@maxikuzmin I am back. I am now on the Pixel 8 Pro instead of the Pixel 6 Pro. I don't have any of the issues so far and can connect to all the WiFis. Maybe the fresh setup helped and it was bug somewhere in my old setup?

Anyway, as of today I also changed my setup slightly:
So far I used Chrome + AdGuard + Cloudflare DNS with AdGuard DNS Filtering (on-device).
I am now using Firefox with uBlock (which has all the AdGuard filters) and excluded Firefox from AdGuard completely.
The reason behind this is, that I don't see a point of filtering the traffic/requests from Firefox twice with the same filters.

I then noticed, that the requests where still being tracked and filters by the AdGuard DNS Filter (on-device) even though Firefox is set to "not route through AdGuard". There doesn't seem to be a way around this. 95% of my traffic/requests come from my browser which already is filtered by uBlock. So I decided to disable AdGuard DNS Filter (on-device) and switched to NextDNS with filtering (remote-filtering). 

Basically AdGuard only has to filter the remaining apps (everything except my browser) which is nearly no traffic. It will probably filter out 99% of those ads on-device and what ever is left (which is nearly nothing) might leave the device (causing data usage which implies battery usage) but will still be blocked by NextDNS. Since it is nearly nothing, I think the data/battery usage can be neglected. But the bonus of not having my huge amount of traffic/requests of my browser being filtered twice is probably going to save some battery!

So this shouldn't make any difference to this issue which means I will keep monitoring the issue and report back!

**Comment by maxikuzmin** [comment]
@Utini2000 congratulations on upgrading your device! In that case, I'm closing this issue for now. If the problem recurs in the future, please reopen the issue. 
If you have problems with filter exclusion for certain applications, please open a new issue

**Comment by Utini2000** [comment]
@maxikuzmin the issue is still present but it seems like it is not as common as before.
How ever, I noticed that when the issue now happens, it also seems to completely freeze the AdGuard app.
Instead of just opening it and disabling / enabling the protection, I now have to kill the app, open it and disable / enable protection.

**Comment by maxikuzmin** [comment]
@Utini2000 please select debug logs mode and wait for the issue to recur, then unload the logs and email [devteam@adguard.com](mailto:devteam@adguard.com) with the time and issue number (4797)

**Comment by maxikuzmin** [comment]
@Utini2000 any news?

**Comment by Utini2000** [comment]
Nothing at the moment. So far it seems to be working again.

**Comment by maxikuzmin** [comment]
@Utini2000 okay, then let's wait a little while longer. Please don't turn off the extended logs. Try to catch this problem again to get the logs

**Comment by maxikuzmin** [comment]
@Utini2000 any news?

**Comment by Utini2000** [comment]
Nothing at the moment but I have also disabled DNS filtering and do it via nextDNS instead.

**Comment by maxikuzmin** [comment]
@Utini2000 as I remember, the problem recurred with the AdGuard DNS filter enabled, and with the AdGuard DNS server? Could you revert back to the configuration where the wifi connection problem was recurring? It would be good to see your logs to get to the root cause of this problem

**Comment by maxikuzmin** [comment]
@Utini2000 any news?

**Comment by Utini2000** [comment]
@maxikuzmin I sent some logs today. I couldn't replicate the issue anymore (it still happens but not as often and not with me being able to force-replicate).

So I completely wiped my phone yesterday.
I imported my old adguard settings/configuration.
Went to my office today and checked what happens.
There was still some initial issue when trying to connect to the wifi but in the end it worked by itself.
Maybe it still helps?
It is not the exact issue I reported here though (where I would only be able to connect by manually enabling/disabling protection).

**Comment by maxikuzmin** [comment]
@Utini2000 didn't quite get it, did you send logs with a record of the bug recurring? At this point, what is the problem with AdGuard? Did you send logs with this problem?

**Comment by maxikuzmin** [comment]
@Utini2000 any news?

**Comment by maxikuzmin** [comment]
@Utini2000  for the moment, I’m closing this issue. You can reopen the issue in the future if the problem persists

**Comment by Utini2000** [comment]
@maxikuzmin alright, I can't really reproduce it ATM. Will report back when I can!

**Comment by Utini2000** [comment]
@maxikuzmin So I just went to the office and noticed that I am not able to connect to the wifi (Android says the WiFi has no internet access).
I open AdGuard and it takes a lot longer than usually to load.
Once AdGuard was open, Android would immediately connect to WiFi.
A few seconds later, I open AdGuard again and it says "restarting protection" and then "protection enabled".

I didn't had debuging enabled (can't run around 24/7 with it as it drains battery) but I still exported logs right after this happend and sent it via mail. 
So the log file was created ~30-60 seconds after the problem occured.

**Comment by maxikuzmin** [comment]
@Utini2000 unfortunately debug logs are needed to get to the bottom of the problem. So far it sounds like you may have unstable wifi in your office. Anyway, please try to catch this with debug logs and send them to us

**Comment by Utini2000** [comment]
@maxikuzmin wifi is definitely not unstable and it also affects several different wifi's. I can see the wifi, it has full signal strength, the phone just won't connect to it until I disable adguard.

I can't run around with debug mode all the time enabled but I will try to catch it.

**Comment by maxikuzmin** [comment]
@Utini2000 hope you can do it. 
It would also be good if you upload the [bug report](https://support.wizyemm.com/hc/en-us/articles/360059933253-How-to-generate-an-Android-bug-report#:~:text=In%20Developer%20options%2C%20tap%20Take,to%20share%20the%20bug%20report) from the Developer options in the phone system when the problem recurs. This will also help to understand what is going on. The logs from AdGuard may not show the nature of the problem.

**Comment by maxikuzmin** [comment]
@Utini2000 any news?

**Comment by Utini2000** [comment]
@maxikuzmin sorry I have AdGuard uninstalled/disabled for now until this major issue gets fixed:
https://github.com/AdguardTeam/AdguardForAndroid/issues/5057

**Comment by maxikuzmin** [comment]
@Utini2000 I hope this problem is solved soon
